addons/mail/static/src/core/web/dialog_patch.js

```diff
@@ -0,0 +1,19 @@
+/* @odoo-module */
+
+import { Dialog } from "@web/core/dialog/dialog";
+import { patch } from "@web/core/utils/patch";
+import { useHotkey } from "@web/core/hotkeys/hotkey_hook";
+
+patch(Dialog.prototype, "mail/views/web", {
+    /**
+     * @override
+     */
+    setup() {
+        this._super();
+        useHotkey("escape", () => {
+            if (this.data.model !== "mail.compose.message") {
+                this.data.close();
+            }
+        });
```

Comment:
Instead of overshadowing `useHotkey` for all dialogs, better override an `onEscape` method.
Just to avoid duplicating the `this.data.close()`